frontend/packages/core/src/stepper.tsx

```diff
@@ -205,19 +228,28 @@ const Stepper = ({ activeStep, orientation = "horizontal", children }: StepperPr
         const stepProps = {
           index: idx + 1,
           variant: "pending" as StepIconVariant,
+          nonLinear,
         };
-        if (idx === activeStep) {
-          stepProps.variant = step.props.error ? "failed" : "active";
-        } else if (idx < activeStep) {
+        const { isComplete = false } = step.props;
+
+        if (isComplete || (!nonLinear && idx < activeStep)) {
           stepProps.variant = "success";
+        } else if (idx === activeStep) {
+          stepProps.variant = step.props.error ? "failed" : "active";
         }
 
+        const label = step.props.label ?? `Step ${idx + 1}`;
+        const icon = <StepIcon {...stepProps} />;
+
         return (
-          <MuiStep key={step.props.label}>
-            {/* eslint-disable-next-line react/no-unstable-nested-components */}
-            <MuiStepLabel StepIconComponent={() => <StepIcon {...stepProps} />}>
-              {step.props.label ?? `Step ${idx + 1}`}
-            </MuiStepLabel>
+          <MuiStep key={label} completed={isComplete}>
+            {nonLinear ? (
```

Comment:
Since `handleStepClick` is optional we should also verify that the function exists before trying to call it. Since the Stepper can be used outside of the Wizard we won't always have a guarantee of it being populated.